multiworld/envs/mujoco/sawyer_reach_torque/sawyer_reach_torque_env.py

```diff
@@ -0,0 +1,417 @@
+
+    def viewer_setup(self):
+        self.viewer.cam.trackbodyid = 0
+        self.viewer.cam.distance = 1.0
+
+        # 3rd person view
+        cam_dist = 0.3
+        rotation_angle = 270
+        cam_pos = np.array([0, 1.0, 0.5, cam_dist, -45, rotation_angle])
+
+        for i in range(3):
+            self.viewer.cam.lookat[i] = cam_pos[i]
+        self.viewer.cam.distance = cam_pos[3]
+        self.viewer.cam.elevation = cam_pos[4]
+        self.viewer.cam.azimuth = cam_pos[5]
+        self.viewer.cam.trackbodyid = -1
+
+    def step(self, a):
+        a = a * self.action_scale
+        self.do_simulation(a, self.frame_skip)
+        if self.use_safety_box:
+            if self.is_outside_box():
+                self.reset_to_prev_qpos()
+            else:
+                self.prev_qpos = self.data.qpos.copy()
+        obs = self._get_obs()
+        info = self._get_info()
+        reward = self.compute_reward(
+            obs['achieved_goal'],
+            obs['desired_goal'],
+            info,
+        )
+        done = False
+        assert reward == -info['hand_distance']
+        return obs, reward, done, info
+
+    def _get_env_obs(self):
+        if self.keep_vel_in_obs:
+            return np.concatenate([
+                self.sim.data.qpos.flat,
+                self.sim.data.qvel.flat,
+                self.get_endeff_pos(),
+            ])
+        else:
+            return np.concatenate([
+                self.sim.data.qpos.flat,
+                self.get_endeff_pos(),
+            ])
+
+    def _get_obs(self):
+        ee_pos = self.get_endeff_pos()
+        state_obs = self._get_env_obs()
+        return dict(
+            observation=state_obs,
+            desired_goal=self._state_goal,
+            achieved_goal=ee_pos,
+
+            state_observation=state_obs,
+            state_desired_goal=self._state_goal,
+            state_achieved_goal=ee_pos,
+        )
+
+    def _get_info(self):
+        hand_distance = np.linalg.norm(self._state_goal - self.get_endeff_pos())
+        # if self.use_goal_caching:
+        #     info =  dict(
+        #         hand_distance=hand_distance,
+        #         hand_success=float(hand_distance < self.indicator_threshold),
+        #     )
+        #     abs_angle_dist = np.abs(self._goal_angles - self._get_env_obs()[:7])
+        #     for i in range(7):
+        #         info['Joint Angle Dim '+str(i+1)] = abs_angle_dist[i]
+        #     return info
+        # else:
+        #     return dict(
+        #         hand_distance=hand_distance,
+        #         hand_success=float(hand_distance < self.indicator_threshold),
+        #     )
+        return dict(
+                hand_distance=hand_distance,
+                hand_success=float(hand_distance < self.indicator_threshold),
+            )
+
+    def get_endeff_pos(self):
+        return self.data.body_xpos[self.endeff_id].copy()
+
+    def reset(self, resample_on_reset=True):
```

Comment:
See slack. Basically, this argument seems redundant given `fix_goal` and `fixed_goal`. If you find yourself using this often, then that's fine, but if not, I'd rather only have one way of specifying whether or not the goals are random/resampled.